### PR TITLE
DMP-2732 Transcript file deletion

### PR DESCRIPTION
**Pull request by DavidTMann**
https://tools.hmcts.net/jira/browse/DMP-2732

Deletion flow for transcript files

Some refactoring between audio/transcript file deletion

Included stringify on dates when passing through state to ensure they persist on refreshes in delete view

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js [hunk 1]
```diff
@@ -139,7 +139,64 @@ describe('Admin - Groups screen', () => {
...
+      cy.get('.govuk-table__row').eq(1).find('.govuk-button.govuk-button--secondary').click();
+
+      cy.get('.govuk-heading-xl').should('contain', 'Delete transcript file');
+
+      cy.get('.govuk-table__header').should('have.length', 8);
+      cy.get('.govuk-table__header').eq(0).should('contain', 'Transcript ID');
+      cy.get('.govuk-table__header').eq(1).should('contain', 'Case ID');
+      cy.get('.govuk-table__header').eq(2).should('contain', 'Hearing date');
+      cy.get('.govuk-table__header').eq(3).should('contain', 'Courthouse');
+      cy.get('.govuk-table__header').eq(4).should('contain', 'Courtroom');
```
Comment (davet1985) [inline cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js:164]: Courtroom isn't a column on the figma and they are also in the wrong order

Screenshot 2024-10-03 at 16 19 56

File: cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js [hunk 2]
```diff
@@ -139,7 +139,64 @@ describe('Admin - Groups screen', () => {
...
+      cy.get('.govuk-table__header').eq(6).should('contain', 'Comments');
+
+      cy.get('.govuk-table__row').should('have.length', 2);
+      cy.get('.govuk-table__row').eq(1).should('contain', '1');
+
+      cy.get('#reject-unhide-option').click();
+
+      cy.get('#confirm-button').click();
+
+      cy.get('#success-message').should('contain', 'Transcript file unmarked for deletion and unhidden');
```
Comment (davet1985) [inline cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js:199]: I'd like to see a test to cover the "You cannot delete a audio file that you marked for deletion." route please.

**Review by davet1985 — CHANGES_REQUESTED** [review]
Couple of minor comments, but looks good 💪

File: cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js (hunk 2 above)
Comment (DavidTMann) [reply cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js:199]: This exists on line 139 I believe Dean done this as part of previous transcript file deletion results screen

File: cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js (hunk 1 above)
Comment (DavidTMann) [reply cypress/e2e/functional/admin-portal/file-deletion-spec.cy.js]: Columns update for delete transcript file view

**Review by davet1985 — COMMENTED** [review]
Good to go for QA

**Review by davet1985 — APPROVED** [review]

**Review by dean-bullock — APPROVED** [review]

**Pull request merged**